File: _files_/_liskov_/liskov.cpp
#include <iostream>

class Madar
{
    
public:
   
    virtual void  repul()
    {
        std::cout << "Repül!\n" ;
    }
};

class Sass : public Madar
{
};

class Pingvin : public Madar
{
};

static void repulMadar(Madar& m)
{
    m.repul();
}

int main()
{
    Sass sass;
    Pingvin pingvin;
    
    repulMadar(sass);
    repulMadar(pingvin);
    
    return 0;
}

File: _files_/_liskov_/Liskov.java
class Madar
{
        
    public void  repul()
    {
			
            System.out.println(this.getClass().getName() + ": " + "repül!");
    }
}
class Sass extends  Madar
{
}
class Pingvin extends  Madar
{
}
class Liskov
{

    public static void repulMadar(Madar madar)
    {
        madar.repul();
    }

    public static void main(String[] args)
    {
        Madar sass = new Sass();
        Madar pingvin = new Pingvin();
        
        repulMadar(sass);
        repulMadar(pingvin);
        
        
    }

}

